### ISSUE TITLE
Fails to push with an empty env In the manifest.

### ISSUE THREAD

**Issue by dfedde** [cloudfoundry/cli#958]
When trying to push a manifest like this:
```
---
disk_quota: 2048M
memory: 2048M
stack: cflinuxfs2
timeout: 180

applications:
- name: web
  buildpack: binary_buildpack
  command: ./app
  health-check-type: port
  instances: 1
  no-route: true
  env:

env:
  VAR: VALUE
```
with ` cf push -f path/to/manifest.yml`

we get:
```
Something unexpected happened. This is a bug in cf.

	Please re-run the command that caused this exception with the environment
	variable CF_TRACE set to true.

	Also, please update to the latest cli and try the command again:
	https://github.com/cloudfoundry/cli/releases

	Please create an issue at: https://github.com/cloudfoundry/cli/issues

	Include the below information when creating the issue:

		Command
		cf push -f ops/cf/manifests/staging.yml

		CLI Version
		6.21.1+6fd3c9f

		Error
		runtime error: invalid memory address or nil pointer dereference

		Stack Trace
			goroutine 1 [running]:
	github.com/cloudfoundry/cli/cf/cmd.generateBacktrace(0x0, 0x0)
		/Users/cffadmin/concourse-work-dir/containers/0000118epdt/tmp/build/254d4083/gopath/src/github.com/cloudfoundry/cli/cf/cmd/cmd.go:185 +0xb4
	github.com/cloudfoundry/cli/cf/cmd.handlePanics(0xc82016aa80, 0x4, 0x4, 0x22c1b28, 0xc8201819b0, 0x22c1ae8, 0xc82026cc80)
		/Users/cffadmin/concourse-work-dir/containers/0000118epdt/tmp/build/254d4083/gopath/src/github.com/cloudfoundry/cli/cf/cmd/cmd.go:169 +0x261
	panic(0x8d9440, 0xc82000e110)
		/usr/local/Cellar/go/1.6.3/libexec/src/runtime/panic.go:443 +0x4e9
	github.com/cloudfoundry/cli/utils/generic.IsMappable(0x0, 0x0, 0xc820406090)
		/Users/cffadmin/concourse-work-dir/containers/0000118epdt/tmp/build/254d4083/gopath/src/github.com/cloudfoundry/cli/utils/generic/map.go:122 +0x120
	github.com/cloudfoundry/cli/utils/generic.mergeReducer(0x77b440, 0xc820406090, 0x0, 0x0, 0x2240950, 0xc8203c6048, 0x0, 0x0)
		/Users/cffadmin/concourse-work-dir/containers/0000118epdt/tmp/build/254d4083/gopath/

		Your Platform Details
		e.g. Mac OS X 10.11, Windows 8.1 64-bit, Ubuntu 14.04.3 64-bit

		Shell
		e.g. Terminal, iTerm, Powershell, Cygwin, gnome-terminal, terminator
```
After removing the env from the web app it works fine.
I would expect for this to work or at least present an error.

**Comment by dkoper** [comment]
Hi @dfedde 

Thanks, we'll look at outputting a proper error message.
The issue is withthe empty `env:`. It's a map, so no value is not a legal value. Use `env: {}` instead.

Regards,
Dies Koper
CF CLI PM

**Comment by dkoper** [comment]
Thanks again for your report.
This has been fixed in the just published 6.22.2 release:
https://github.com/cloudfoundry/cli#downloads

Cheers,
Dies Koper
CF CLI PM
